src/alphafold_impact/pipelines/data_collection_nih_funded_projects/nodes.py

```diff
@@ -0,0 +1,174 @@
+"""
+This module contains functions for fetching NIH funded projects data.
+
+Functions:
+    collect_nih_funded_projects: Downloads and saves NIH funded projects
+        to S3 as a partitioned dataset.
+"""
+
+import logging
+import time
+from typing import List, Dict, Tuple, Callable
+from datetime import datetime
+from calendar import monthrange
+import requests
+from requests.adapters import HTTPAdapter, Retry
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+
+def _get_month_range(year: int, month: int) -> Tuple[str, str]:
+    """
+    Returns the first and last day of a given month in a specified year.
+
+    Args:
+        year (int): The year.
+        month (int): The month.
+
+    Returns:
+        Tuple[str, str]: A tuple containing the first and last day of the month.
+    """
+    first_day = datetime(year, month, 1)
+    _, last_day_num = monthrange(year, month)
+    last_day = datetime(year, month, last_day_num)
+    return first_day.strftime("%Y-%m-%d"), last_day.strftime("%Y-%m-%d")
+
+
+def _setup_session() -> requests.Session:
+    """Set up the requests session with retry strategy."""
+    session = requests.Session()
+    retries = Retry(
+        total=5,
+        backoff_factor=0.3,
+        status_forcelist=[429, 500, 502, 503, 504],
+        allowed_methods=["POST"],
+    )
+    session.mount("https://", HTTPAdapter(max_retries=retries))
+    return session
+
+
+def _get_nih_projects(session: requests.Session, query_params: dict) -> dict:
```

Comment:
Since we're generally sticking to typing.. 
```suggestion
def _get_nih_projects(session: requests.Session, query_params: Dict[str, str]) -> Dict[str, str]:
```